### PR TITLE
bunch of fixes

### PR DESCRIPTION
**Pull request by bam241**
Allows to build a conda recipe for cyclus
Also address deprecation from boost 1.68 also.

change in `src/OsiCbcSolverInterface.cpp` were required to build on `condaforge/linux-anvil-comp7`, I don't know if it will break the linux build... 

the change in `cycpp.py` and in `UseCyclus` should be transparent unless you have a `$CPP` variable setup on your system, if so `cmake` will use this value in place of `cpp` executable...

### CODE REVIEW

**Comment by bam241** [comment]
@katyhuff @gonuke @FlanFlanagan could you look into this, I probably need this to fix the conda builds :)

File: cmake/UseCyclus.cmake [hunk 1]
```diff
@@ -89,11 +89,15 @@ MACRO(USE_CYCLUS lib_root src_root)
 
 
     # set cpp path
-    IF("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
-        SET(PREPROCESSOR "--cpp-path=clang++")
+    
+    IF(DEFINED ENV{CPP})
```
Comment (gonuke) [inline cmake/UseCyclus.cmake]: add a comment to explain what's happening here

Comment (bam241) [reply cmake/UseCyclus.cmake]: @gonuke does it look better ?

**Comment by FlanFlanagan** [comment]
Looks good to me.

**Comment by bam241** [comment]
ping @gonuke

File: cmake/UseCyclus.cmake (hunk 1 above)
Comment (katyhuff) [reply cmake/UseCyclus.cmake]: lgtm

**Pull request merged**